### PR TITLE
Moto server implementation for pytest

### PR DESCRIPTION
**Pull request by sebpretzer**
This is the PR relating to https://github.com/apache/iceberg-python/issues/222. 

This should be a drop in replacement for `moto.mock_s3`, by pointing the s3 clients at a moto server endpoint url. 

By doing this, you can:
1. Remove the patch to `aiobotocore`
2. Remove the requirement of using `pyiceberg.io.fsspec.FsspecFileIO` over the default of `pyiceberg.io.pyarrow.PyArrowFileIO`.

Glue and DynamoDB still use the basic `mock_glue` and `mock_dynamodb`. They could probably be moved over to the sever implementation as well, but it was taking more work than I wanted so I didn't include it.

### CODE REVIEW

**Comment by Fokko** [comment]
cc @HonahX since he's the original author of the Glue catalog

File: tests/conftest.py [hunk 1]
```diff
@@ -1661,11 +1586,33 @@ def fixture_aws_credentials() -> Generator[None, None, None]:
     os.environ.pop("AWS_DEFAULT_REGION")
 
 
+MOTO_SERVER = ThreadedMotoServer(port=5456)
```
Comment (HonahX) [inline tests/conftest.py:1589]: Just out of curiosity, why do we go with port '5456'? Is there a particular reason or was it more of a random pick?

**Review by HonahX — APPROVED** [review]
LGTM! Thanks for the fix @sebpretzer! It is great that we can finally get rid of the `_patch_aiobotocore` and use `PyarrowFileIO` in the unit tests!

File: tests/conftest.py (hunk 1 above)
Comment (sebpretzer) [reply tests/conftest.py:1589]: No particular reason. The [default is 5000](https://github.com/getmoto/moto/blob/a8d7dfd7af198838152687dd80867653e6e5126a/moto/moto_server/threaded_moto_server.py#L12). I think I had something running at 5000 when I had initially set this up. Let me drop it in favor of the default.

Comment (sebpretzer) [reply tests/conftest.py]: Fixed!

Aside: I actually tracked down what was [actually happening](https://nono.ma/port-5000-used-by-control-center-in-macos-ventura-controlce), it was not something I had stood up at 5000... 🤦‍♂️

**Review by Fokko — APPROVED** [review]
Thanks for working on this @sebpretzer 🙌  and thanks for the review @HonahX

**Pull request merged**